Update index error handling

Roang-zero1 opened the pull request:
I've updated the error handling for invalid indexes.
The new version is compatible with the types I will be submitting shortly.

Error messages are now narrowed to the specific error.

<!--

Read the [contributing guidelines](https://github.com/syntax-tree/.github/blob/master/contributing.md).

We are excited about pull requests, but please try to limit the scope, provide a general description of the changes, and remember, it's up to you to convince us to land it.

If this fixes an open issue, link to it in the following way: `Closes GH-123`.

New features and bug fixes should come with tests.

P.S. have you seen our support and contributing docs?
https://github.com/syntax-tree/.github/blob/master/support.md
https://github.com/syntax-tree/.github/blob/master/contributing.md
-->

wooorm commented:
I think this will change behaviour of e.g., falsey values that are not a number are passed through.
Is there a reason for changing the behaviour, other than error messages?

Roang-zero1 commented:
I've updated the error handling to include falsey values.
The reason I updated the error handling is to be able to use typescript to add the types in #7 .
The original implementation is not typescript compatible, so I've updated it.

wooorm commented on index.js:
```diff
@@ -18,12 +18,18 @@ function findAllAfter(parent, index, test) {
   length = children.length
 
-  if (index && index.type) {
+  if (index === undefined) {
```
```suggestion
  if (index === undefined || index === null) {
```

`null` and `undefined` are typically treated the same, so I think for them to have the same error message here makes sense.

wooorm commented on index.js:
```diff
@@ -18,12 +18,18 @@ function findAllAfter(parent, index, test) {
+    throw new Error('Expected positive finite index or child node')
+  } else if (index && typeof index !== 'number') {
     index = children.indexOf(index)
+    if (index < 0) {
```
```suggestion
    if (index === -1) {
```

`index < 0` can only be one value, this is more explicit.

wooorm commented on index.js:
```diff
@@ -18,12 +18,18 @@ function findAllAfter(parent, index, test) {
-  if (isNaN(index) || index < 0 || index === Infinity) {
-    throw new Error('Expected positive finite index or child node')
+  if (typeof index !== 'number' || index < 0 || index === Infinity) {
+    console.log('Throw finite index')
```
```suggestion
```

Roang-zero1 commented:
Thanks for the suggestions, I've committed them.

wooorm commented:
Released!
